### ISSUE TITLE
Issue with prediction

### ISSUE THREAD

**Issue by Rakshith-Manandi** [jakeret/tf_unet#197]
The model is training with 99% accuracy but there's no prediction. I assume there's overfitting happening and I needed some guidance overcoming this issue. A brief gist of what I've done so far:
I've used the SimpleDataProvider class from the image_util.py file to generate my images. In the BaseProviderClass under the process_labels function you use the ~label which was converting my data to -1 and 0. 
1) I assume you're doing this to onehotencode the labels. I used np.logical_not(label) and now my data is 0 & 1. Is this correct?
2) My model is training at 99% accuracy but there's no prediction happening. What could be the problem? I'm training on 165 images for 10 epochs and 32 iterations each. Should I increase my epochs given that my training data is high? I tried increasing the layers to 5 from 3 and I got like a weird output, which is what makes me think that there's something wrong with what I'm doing and maybe the model didn't train at 99% when I had 3 layers. 

Any help on this is appreciated! Thanks

**Comment by Rakshith-Manandi** [comment]
Also, can you please advise on what should be given in the cost_kwargs argument to make use of the regularizer. I tried cost_kwargs = {"regularizer":[l2]} and I got name l2 not defined. 

Thanks,
Rakshith.

**Comment by jakeret** [comment]
Before using a deeper architecture I would rather try to increase the number of filters (`features_root=XX`).
Have you checked the output in the tensorboard. Is the loss decreasing? Do the other metrics look "reasonable"? 
Could it be that your data is very unbalanced i.e. 99% of the pixel belong to one class?
As for the regularizer: the implementation expects the power of the l2 regularizer i.e. a float e.g. 0.001

**Comment by Rakshith-Manandi** [comment]
I'll try increasing the number of filters. Thanks.

The loss is decreasing but I'm still not getting the output. Your guess is correct, 99% of the pixels belong to one class. How do I overcome this?

**Comment by Rakshith-Manandi** [comment]
I tried both 'dice_coefficient' and class weights, didn't seem to have much luck. I'll try the synthetic data method and see if I'm getting any results. Thank you for the response :)

**Comment by Rakshith-Manandi** [comment]
With dice_coefficient I'm getting negative minibatch loss. Does that seem right to you?

**Comment by jakeret** [comment]
Yes that's fine. It's because SGD is minimizing the loss and a large dice corresponds to a great overlap, hence * -1
